okhttp-clients/src/test/java/com/palantir/conjure/java/okhttp/InstrumentedInterceptorTest.java

```diff
@@ -107,6 +112,21 @@ public void testIoExceptionRecorded() throws IOException {
         assertThat(metrics.getIoExceptions().getCount()).isEqualTo(1);
     }
 
+    @Test
+    public void testIoExceptionNotRecordedWhenCancelled() throws IOException {
+        when(chain.request()).thenReturn(REQUEST_A);
+        when(chain.proceed(any())).thenThrow(IOException.class);
+        when(chain.call()).thenReturn(call);
+        when(call.isCanceled()).thenReturn(true);
```

Comment:
may be nice to have a `failedRequest(bool cancelled)` in a similar way to the `successfulRequest` method which does all this setup.